torch/cuda/__init__.py

```diff
@@ -467,8 +467,11 @@ class _CudaBase(object):
     is_sparse = False
 
     def type(self, *args, **kwargs):
-        with device(self.get_device()):
-            return super(_CudaBase, self).type(*args, **kwargs)
+        # We could use a Protocol here to tell mypy that self has `get_device` method
+        # but it is only available on Python >= 3.8 on typing or mypy_extensions on
```

Comment:
I think you mean `typing_extensions`? That is a non-optional dependency already, so could be used. That said, I don't think it's necessary to spend time on this now. Just updating the comment may be good for now.